packages/user-interface/src/components/ExtraCaseDetails.tsx

```diff
@@ -0,0 +1,151 @@
+    typeof obj.rows === "object"
+  );
+};
+
+const parseDetails = (details: Details[]) => {
+  const zaakDetailsContent: ReactNode[] = [];
+
+  details.forEach((detail) => {
+    const headingElement = detail.heading ? (
+      <SectionHeader title={detail.heading} />
+    ) : undefined;
+    const waarde = detail.waarde;
+
+    // Waarde type string[]
+    if (hasStringWaardes(waarde)) {
+      zaakDetailsContent.push(
+        <section>
+          {headingElement}
+          <UnorderedList>
+            {waarde.map((item, index) => (
+              <UnorderedListItem key={index}>{item}</UnorderedListItem>
+            ))}
+          </UnorderedList>
+        </section>,
+      );
+    }
+
+    // Waarde type Details (and Table)
+    if (hasDetailsWaardes(waarde)) {
+      const descriptionListItems = waarde.filter(
+        (item) => item.type !== "table",
+      );
+      const tableItems = waarde.filter((item) => item.type === "table");
+
+      const dlItems = descriptionListItems
+        .filter(
+          (item) =>
+            typeof item.waarde === "string" || typeof item.waarde === "boolean",
+        )
+        .map((item) => {
+          return { title: item.key, detail: item.waarde as string };
+        });
+
+      zaakDetailsContent.push(
+        <section>
+          {headingElement}
+          <DescriptionList items={dlItems} />
+        </section>,
+      );
+
+      tableItems
+        .filter((item) => isTable(item.waarde))
+        .forEach((tableDetail) => {
+          const table = tableDetail.waarde as Table;
+          zaakDetailsContent.push(
+            <section>
+              <TableList
+                key={tableDetail.heading}
+                titleTranslationId={tableDetail.heading}
+                headers={table.headers?.map((head) => head.waarde.toString())}
+                rows={table.rows.map((row) =>
+                  row.map((cell) => cell.waarde as CellObject),
+                )}
+              />
+            </section>,
+          );
+        });
+    }
+
+    // Waarde types string | boolean
+    if (typeof waarde === "string" || typeof waarde === "boolean") {
+      zaakDetailsContent.push(
+        <section>
+          {headingElement}
+          <DescriptionList
+            items={[
+              {
+                title: detail.key,
+                detail:
+                  typeof waarde === "string" ? (
+                    waarde
+                  ) : (
+                    <FormattedMessage
+                      id={
+                        waarde
+                          ? "extraCaseDetails.boolean.true"
+                          : "extraCaseDetails.boolean.false"
+                      }
+                    />
+                  ),
+              },
+            ]}
+          />
+        </section>,
+      );
+    }
+  });
+
+  return zaakDetailsContent;
+};
+
+export const ExtraCaseDetails = ({ data }: Props) => {
+  return parseDetails(data);
```

Comment:
Waarom is dit een aparte functie?

Reply:
Ik had eerst een oplossing op een recursieve manier gemaakt, waarin de nesting goed zou moeten werken, maar dat werkte uiteindelijk niet, vandaar dat het nog in een aparte functie zit. Zal het verplaatsen naar het component zelf.